x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdate.java

```diff
@@ -179,6 +187,9 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalLong(modelSnapshotRetentionDays);
         out.writeOptionalLong(dailyModelSnapshotRetentionAfterDays);
         out.writeOptionalLong(resultsRetentionDays);
+        if (out.getVersion().onOrAfter(Version.CURRENT)) {  // TODO: 7.15
+            out.writeOptionalLong(annotationsRetentionDays);
```

Comment:
I'm wondering if at some point we'd like to encode some special value, e.g.: "unset" as -1.

Reply:
I mean, we could at some point, but since we haven't done that for any of the other values yet (which have been around for years), I think it's not a big concern...

But, if you really don't want to serialize with VLong, the cost difference is not big.